newIDE/app/src/MainFrame/EditorContainers/HomePage/BuildSection/utils.js

```diff
@@ -108,81 +112,166 @@ export const transformCloudProjectsIntoFileMetadataWithStorageProviderName = (
+              onSelectExample: onSelectExampleShortHeader,
+            })
+      );
+    }
+    if (i >= carouselExclusiveItemsCount) {
+      if (shouldAddPrivateGameTemplate) {
+        const privateGameTemplateListingData =
+          privateGameTemplateListingDatas[privateGameTemplateIndex];
+        if (privateGameTemplateListingData) {
           const isTemplateOwned =
-            !!authenticatedUser.receivedGameTemplates &&
-            !!authenticatedUser.receivedGameTemplates.find(
+            !!receivedGameTemplates &&
+            !!receivedGameTemplates.find(
               receivedGameTemplate =>
                 receivedGameTemplate.id === privateGameTemplateListingData.id
             );
-          return {
-            id: privateGameTemplateListingData.id,
-            title: privateGameTemplateListingData.name,
-            thumbnailUrl: privateGameTemplateListingData.thumbnailUrls[0],
-            onClick: () => {
-              sendGameTemplateInformationOpened({
-                gameTemplateName: privateGameTemplateListingData.name,
-                gameTemplateId: privateGameTemplateListingData.id,
-                source: 'homepage',
-              });
-              onSelectPrivateGameTemplateListingData(
-                privateGameTemplateListingData
-              );
-            },
-            overlayText: getProductPriceOrOwnedLabel({
-              i18n,
-              productListingData: privateGameTemplateListingData,
-              owned: isTemplateOwned,
-            }),
-            overlayTextPosition: 'topLeft',
-          };
-        })
-      : []),
-  ];
-
-  const exampleShortHeaderItems = [
-    ...(exampleShortHeaders
-      ? prepareExampleShortHeaders(exampleShortHeaders)
-          .map(example => ({
-            id: example.id,
-            title: example.name,
-            onClick: () => onSelectExampleShortHeader(example),
-            thumbnailUrl: example.previewImageUrls[0],
-          }))
-          .filter(exampleShortHeader => !!exampleShortHeader.thumbnailUrl)
-      : []),
-  ];
-
-  for (let i = 0; i < exampleShortHeaderItems.length; ++i) {
-    allItems.push(exampleShortHeaderItems[i]);
-    if (i % 2 === 1 && privateGameTemplateItems.length > 0) {
-      const nextPrivateGameTemplateIndex = Math.floor(i / 2);
-      if (nextPrivateGameTemplateIndex < privateGameTemplateItems.length)
-        allItems.push(privateGameTemplateItems[nextPrivateGameTemplateIndex]);
+          gridItems.push(
+            <PrivateGameTemplateTile
+              privateGameTemplateListingData={privateGameTemplateListingData}
+              onSelect={() => {
+                onSelectPrivateGameTemplateListingData(
+                  privateGameTemplateListingData
+                );
+              }}
+              owned={isTemplateOwned}
+              key={privateGameTemplateListingData.id}
+            />
+          );
+          privateGameTemplateActuallyAdded = true;
```

Comment:
I haven't looked at the whole logic, but this kind of prop feels... like a hack and could be done in a cleaner way (but I may be wrong)

Reply:
If it works all good, it just felt like a code smell :)